src/components/MapVizzard/MapLayers/MapLayers.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import styles from './MapLayers.module.css';
+import LayerRow from './LayerRow';
+
+function MapLayers({
+  layers,
+  setLayers,
+  activeLayer,
+  setActiveLayer,
+}) {
+  const updateLayers = (d, id) => {
+    const layersClone = [...layers];
+    if (d === 'remove') {
+      const newLayersObj = layersClone.filter((obj) => obj.id !== id);
+      if (setLayers) setLayers([...newLayersObj]);
+    } else {
+      layersClone.forEach((dd, ii) => {
+        if (dd.id === id) {
+          layersClone[ii] = d;
+        }
+      });
```

Comment:
It's better to use Array.findIndex to get the index and then update the layersClone.
This makes the intent very clear.

```
const layerIndex = layersClone.find((layer) => layer.id === id);
if (layerIndex !== -1) {
    layersClone[layerIndex] = d;
}
```